lib/rust/frp/src/network.rs

```diff
@@ -99,25 +103,41 @@ impl Network {
     /// network, like animation instances.
     pub fn store<T: 'static + CloneRef>(&self, item: &T) {
         let item = item.clone_ref();
-        self.data.storage.borrow_mut().push(Box::new(item));
+        self.store_boxed(Box::new(item));
+    }
+
+    /// Store an arbitrary boxed item.
+    ///
+    /// Force the compiler to never inline this function, so the generated code (including two panic
+    /// handlers) can be shared across all registered items independent of their type.
+    #[inline(never)]
+    fn store_boxed(&self, item: Box<dyn Any>) {
+        self.data.storage.borrow_mut().push(item);
     }
```

Comment:
Does it affect performance of FRP? Did you check it? This will be called a lot of times often in performance heavy places.

Reply:
I did check those "_boxed" methods together back when I introduced them. It did have some impact, but I don't have the numbers saved. It would now require new separate measurement to quantify how important it is. It might also have an impact on final binary size, so both would need to be checked.

Reply:
Our FRP is not the fastest implementation possible, unfortunately. And thus I care about performance of it as its used in multiple performance critical sections of the renderer. If this affected performance even slightly, I think we should leave it as is and improve FRP with the new implementation in the future.

Reply:
Looking at the code again, this specific function (`store`) is not actually called with massive amount of types, so avoiding monomorphization here probably doesn't have any significant effect. The `register` one is definitely the more important one in that regard. I changed them together because it's the same pattern, so I didn't even considered if it's necessary on both.

Still, granted I don't have any hard data to back it up right now, I don't see any reason why this change would affect perfrormance negatively in any way. Neither `borrow_mut` nor push can be simplified by inlining in our usage patterns, as `borrow_mut` check must always be performed and `Vec` is never preallocated to even have a chance of avoiding the bound check/grow path. If anything, having the code not duplicated might help a little due to better instruction cache/branch predictor utilization. But this is really splitting hairs, as we have much bigger and more fundamental problems with performance, as you already mentioned. In reality it probably doesn't have any noticeable effect and I'd just call this change unnecessary. I'm ok with reverting that tiny portion if you feel it's safer that way.